packages/office-ui-fabric-react/src/components/Button/Button.test.tsx

```diff
@@ -640,92 +640,135 @@ describe('Button', () => {
         });
         expect(didClick).toEqual(false);
       });
-    });
 
-    describe('with contextual menu', () => {
-      function buildRenderAndClickButtonAndReturnContextualMenuDOMElement(
-        menuPropsPatch?: any,
-        text?: string,
-        textAsChildElement: boolean = false
-      ): HTMLElement {
-        const menuProps = { items: [{ key: 'item', name: 'Item' }], ...menuPropsPatch };
-        const element: React.ReactElement<any> = (
+      it('A disabled Button does not respond to input events', () => {
+        const renderedDOM: HTMLElement = renderIntoDocument(
           <DefaultButton
-            iconProps={{ iconName: 'Add' }}
-            text={!textAsChildElement && text ? text : undefined}
-            menuProps={menuProps}
-          >
-            {textAsChildElement && text ? text : null}
-          </DefaultButton>
+            disabled={true}
+            data-automation-id="test"
+            text="Create account"
+            split={false}
+            onClick={setTrue}
+            onKeyPress={setTrue}
+            onKeyUp={setTrue}
+            onKeyDown={setTrue}
+            onMouseDown={setTrue}
+            onMouseUp={setTrue}
+          />
         );
 
-        const button = ReactTestUtils.renderIntoDocument<any>(element) as React.ReactInstance;
-        const renderedDOM = ReactDOM.findDOMNode(button) as HTMLElement;
-
-        expect(renderedDOM).toBeDefined();
         ReactTestUtils.Simulate.click(renderedDOM);
+        ReactTestUtils.Simulate.keyDown(renderedDOM, {
+          which: KeyCodes.down,
+          altKey: true
+        });
+        ReactTestUtils.Simulate.keyUp(renderedDOM, {
+          which: KeyCodes.down,
+          altKey: true
+        });
+        ReactTestUtils.Simulate.keyPress(renderedDOM, {
+          which: KeyCodes.down,
+          altKey: true
+        });
+        ReactTestUtils.Simulate.mouseDown(renderedDOM, {
+          type: 'mousedown',
+          clientX: 0,
+          clientY: 0
+        });
+
+        ReactTestUtils.Simulate.mouseUp(renderedDOM, {
+          type: 'mouseup',
+          clientX: 0,
+          clientY: 0
+        });
+        // expect(didClick).toEqual(false);
```

Comment:
This shouldn't be commented out, currently this test isn't testing anything, right?

Also, what about testing when allowDisabledFocus is true vs false for buttons, buttons that have menus, and splitButtons (and splitButtons have primaryDisabled in addition to disabled)?